src/components/MessageList/MessageList.tsx

```diff
@@ -825,36 +830,57 @@ const MessageListWithContext = <
   const onScrollEndDrag = () =>
     hasMoved && selectedPicker && setHasMoved(false);
 
+  const refCallback = (
+    ref: FlatListType<MessageType<At, Ch, Co, Ev, Me, Re, Us>>,
+  ) => {
+    flatListRef.current = ref;
+
+    if (setFlatListRef) {
+      setFlatListRef(ref);
+    }
+  };
+
+  const renderFooterComponent = () => (
+    <FooterComponent loadingMore={onEndReachedInProgress} />
+  );
+
+  const renderHeaderComponent = () => (
+    <HeaderComponent loadingMore={onStartReachedInProgress} />
+  );
+
+  const renderEmptyStateIndicator = () => (
+    <View style={styles.flex}>
+      <View style={styles.flex} testID='empty-state'>
+        <EmptyStateIndicator listType='message' />
+      </View>
+    </View>
+  );
+
   if (!FlatList) return null;
+
+  if (loading) {
+    return (
+      <View style={styles.flex}>
+        <LoadingIndicator listType='message' />
+      </View>
+    );
+  }
+
   return (
     <View
       style={[styles.container, { backgroundColor: white_snow }, container]}
     >
       <FlatList
         contentContainerStyle={styles.contentContainer}
         data={messageList}
-        /** Disables the MessageList UI. Which means, message actions, reactions won't work. */
         extraData={disabled || !channel?.state.isUpToDate}
+        /** Disables the MessageList UI. Which means, message actions, reactions won't work. */
```

Comment:
should this comment still be above extraData?

Reply:
Done!!